### ISSUE TITLE
500 error if BASE_DIR not set

### ISSUE THREAD

**Issue by pasevin** [ImperialCollegeLondon/django-drf-filepond#18]
Hi I think that bugfix on issue #5 didn't solve the problem. Unless I'm doing something cardinally wrong.

```
        if ((not hasattr(local_settings, 'UPLOAD_TMP')) or
                (not (storage.location).startswith(local_settings.BASE_DIR))):
            return Response('The file upload path settings are not '
                            'configured correctly.',
                            status=status.HTTP_500_INTERNAL_SERVER_ERROR)
```

Check fails on  ```not (storage.location).startswith(local_settings.BASE_DIR)```

Debugger variables:
```storage.location```:  ```/var/www/domain/proj_name/temp/filepond_uploads```
```local_settings.BASE_DIR```: ```/var/www/domain/venv/lib/python3.6/site-packages/django_drf_filepond```

**Comment by jcohen02** [comment]
Hi @pasevin, thanks for highlighting this. Not sure what happened here but you're correct that this still seems to be an issue.

I'll get a fix for this created shortly. If you'd be able to test it, that would be great.

**Comment by pasevin** [comment]
Hi, yes sure I will test!
I don't have ```BASE_DIR``` set, but when I do set it, the library works without issue.
And yes your'e correct, when ```BASE_DIR``` is not set, then the lib defaults to the venv destination where it's installed.

Sorry for the confusion, ```proj_name``` is just a subdirectory of the project where ```settings.py``` and static directory is situated. This is where I wanted to keep my temp uploads.  ```BASE_DIR``` is ```/var/www/domain/```.

**Comment by jcohen02** [comment]
@pasevin I've pushed a fix to the [fix/issue18](/ImperialCollegeLondon/django-drf-filepond/tree/fix/issue18) branch. If you have a chance to test this and check that it does indeed resolve the issue you were experiencing, that would be great but I'll merge in the next couple of days anyway. The updated tests are passing. 

One thing to note, if you have `BASE_DIR` set but are setting a storage location outside of `BASE_DIR`, you need to set the `DJANGO_DRF_FILEPOND_ALLOW_EXTERNAL_UPLOAD_DIR` parameter to `True` in your settings.

This was intended to ensure that the developer things about any potential security implications with allowing the app to store arbitrary temporary uploads at a location outside the file tree of the deployed application. I opted to not bother with requiring this flag if you don't have `BASE_DIR` set, but maybe is should be required in that case too?

Thanks.

**Comment by pasevin** [comment]
I've tried removing the BASE_DIR and all works fine.

Re: ```DJANGO_DRF_FILEPOND_ALLOW_EXTERNAL_UPLOAD_DIR``` I'm not sure if this is necessary at all. Maybe just a warning would suffice on init if location has been set outside the BASE_DIR.

This is for you to decide :)

**Comment by jcohen02** [comment]
Thanks for checking this and glad it's working ok.
Not sure what to do with the `DJANGO_DRF_FILEPOND_ALLOW_EXTERNAL_UPLOAD_DIR` attribute. As you say, I'm inclined to think it's probably not necessary but since it's implemented I may leave it in there for now, we can always revert to a warning subsequently.

I'll get this merged in asap.

**Comment by jcohen02** [comment]
Merged and included in release 0.2.2.
